### PR TITLE
feature: add function get_nbasis in class HContainer

### LINKED ISSUE #4393: Can not get the matrix dimension of HContainer
### Describe the Code Quality Issue

We need an interface to get the dimension of the dense matrix for HContainer.

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Identify the specific code file or section with the code quality issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified issue.
- [ ] Refactor the code to improve code quality, following the suggested solution.
- [ ] Ensure the refactored code adheres to the project's coding standards.
- [ ] Test the refactored code to ensure it functions as expected.
- [ ] Update any relevant documentation, if necessary.
- [ ] Submit a pull request with the refactored code and a description of the changes made.

### PR DESCRIPTION
**Pull request by pxlxingliang**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #4393

### Unit Tests and/or Case Tests for my changes
- A unit test is added for each new feature or bug fix.

### What's changed?
- Example: My changes might affect the performance of the application under certain conditions, and I have tested the impact on various scenarios...

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

File: source/module_hamilt_lcao/module_hcontainer/hcontainer.h [hunk 1]
```diff
@@ -404,12 +404,17 @@ class HContainer
...
      * @brief get sparse_ap_index
      * @return std::vector<std::vector<int>>&
-    */
+     */
     const std::vector<std::vector<int>>& get_sparse_ap_index() const
     {
         return sparse_ap_index;
     }
 
+    const Parallel_Orbitals* get_paraV() const
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_hcontainer/hcontainer.h]: Let Hcontainer explicitly depends on Parallel_Orbitals, do we really need this dependency?

Comment (pxlxingliang) [reply source/module_hamilt_lcao/module_hcontainer/hcontainer.h]: I think that it depends on the positioning of the Parallel-Orbitals class. If we consider it as a basic class for managing data parallelism information in ABACUS, I think there is no problem returning this member in HContainer.

Comment (mohanchen) [reply source/module_hamilt_lcao/module_hcontainer/hcontainer.h]: Could you discuss with Daye?

Comment (pxlxingliang) [reply source/module_hamilt_lcao/module_hcontainer/hcontainer.h]: After discussed with Daye, we think that it is better to add a function to return the basis number directly.

**Review by mohanchen — APPROVED** [review]

**Pull request merged**